samples/boards/sensortile_box_pro/sensors-on-board/src/main.c

```diff
@@ -0,0 +1,301 @@
+	const struct device *const lis2mdl = DEVICE_DT_GET_ONE(st_lis2mdl);
+
+	if (!device_is_ready(hts221)) {
+		printk("%s: device not ready.\n", hts221->name);
+		return 0;
+	}
+	if (!device_is_ready(lsm6dsv16x)) {
+		printk("%s: device not ready.\n", lsm6dsv16x->name);
+		return 0;
+	}
+	if (!device_is_ready(lis2mdl)) {
+		printk("%s: device not ready.\n", lis2mdl->name);
+		return 0;
+	}
+
+	lis2mdl_config(lis2mdl);
+	lsm6dsv16x_config(lsm6dsv16x);
+
+	while (1) {
+		struct sensor_value hts221_hum, hts221_temp;
+		struct sensor_value lsm6dsv16x_accel[3], lsm6dsv16x_gyro[3];
+		struct sensor_value lis2mdl_magn[3];
+		struct sensor_value lis2mdl_temp;
+
+		/* handle HTS221 sensor */
+		if (sensor_sample_fetch(hts221) < 0) {
+			printf("HTS221 Sensor sample update error\n");
+			return 0;
+		}
+
+#ifndef CONFIG_LSM6DSV16X_TRIGGER
+		if (sensor_sample_fetch(lsm6dsv16x) < 0) {
+			printf("LSM6DSV16X Sensor sample update error\n");
+			return 0;
+		}
+#endif
+
+#ifndef CONFIG_LIS2MDL_TRIGGER
+		if (sensor_sample_fetch(lis2mdl) < 0) {
+			printf("LIS2MDL Magn Sensor sample update error\n");
+			return 0;
+		}
+#endif
+
+		sensor_channel_get(hts221, SENSOR_CHAN_HUMIDITY, &hts221_hum);
+		sensor_channel_get(hts221, SENSOR_CHAN_AMBIENT_TEMP, &hts221_temp);
+		sensor_channel_get(lsm6dsv16x, SENSOR_CHAN_ACCEL_XYZ, lsm6dsv16x_accel);
+		sensor_channel_get(lsm6dsv16x, SENSOR_CHAN_GYRO_XYZ, lsm6dsv16x_gyro);
+		sensor_channel_get(lis2mdl, SENSOR_CHAN_MAGN_XYZ, lis2mdl_magn);
+		sensor_channel_get(lis2mdl, SENSOR_CHAN_DIE_TEMP, &lis2mdl_temp);
+
+		/* Display sensor data */
+
+		/*  Clear terminal (ANSI ESC-C) */
+		printf("\0033\014");
+
+		printf("SensorTile.box dashboard\n\n");
+
+		/* HTS221 temperature */
+		printf("HTS221: Temperature: %.1f C\n",
+		       sensor_value_to_double(&hts221_temp));
+
+		/* HTS221 humidity */
+		printf("HTS221: Relative Humidity: %.1f%%\n",
+		       sensor_value_to_double(&hts221_hum));
+
+		printf("LSM6DSV16X: Accel (m.s-2): x: %.3f, y: %.3f, z: %.3f\n",
+			sensor_value_to_double(&lsm6dsv16x_accel[0]),
+			sensor_value_to_double(&lsm6dsv16x_accel[1]),
+			sensor_value_to_double(&lsm6dsv16x_accel[2]));
+
+		printf("LSM6DSV16X: GYro (dps): x: %.3f, y: %.3f, z: %.3f\n",
+			sensor_value_to_double(&lsm6dsv16x_gyro[0]),
+			sensor_value_to_double(&lsm6dsv16x_gyro[1]),
+			sensor_value_to_double(&lsm6dsv16x_gyro[2]));
+
+		/* lis2mdl */
+		printf("LIS2MDL: Magn (gauss): x: %.3f, y: %.3f, z: %.3f\n",
+		       sensor_value_to_double(&lis2mdl_magn[0]),
+		       sensor_value_to_double(&lis2mdl_magn[1]),
+		       sensor_value_to_double(&lis2mdl_magn[2]));
+
+		printf("LIS2MDL: Temperature: %.1f C\n",
+		       sensor_value_to_double(&lis2mdl_temp));
+
+#ifdef CONFIG_LSM6DSV16X_TRIGGER
+		printk("%d:: lsm6dsv16x acc trig %d\n", cnt, lsm6dsv16x_acc_trig_cnt);
+		printk("%d:: lsm6dsv16x gyr trig %d\n", cnt, lsm6dsv16x_gyr_trig_cnt);
+#endif
+
+#if defined(CONFIG_LIS2MDL_TRIGGER)
```

Comment:
nit: the rest of the files uses `#ifdef` rather than `#if defined`, may as well use `#ifdef` here and keep it coherent

Reply:
ack